In [20]:
import pandas as pd
import matplotlib.pyplot as plt

from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import r2_score, mean_absolute_error
from sklearn.preprocessing import StandardScaler

In [21]:
df = pd.read_excel("yigitdsa210data.xlsx")
df.head()

,Player,Nation,Pos,Squad,Age,Born,MP,Starts,Min,Gls,Ast,G+A,Gls.1,Ast.1,Market Value
0,Miguel Almirón,py PAR,FW,Newcastle United,30.0,1994.0,9.0,1,157.000,0.0,0.0,0.0,0.00,0.00,€18.00m
1,Michail Antonio,jm JAM,FW,West Ham United,34.0,1990.0,14.0,11,836.000,1.0,1.0,2.0,0.11,0.11,€6.00m
2,Antony,br BRA,"FW,MF",Manchester Utd,24.0,2000.0,8.0,0,141.000,0.0,0.0,0.0,0.00,0.00,€40.00m
3,Cameron Archer,eng ENG,FW,Southampton,23.0,2001.0,35.0,13,1.441,2.0,0.0,2.0,0.12,0.00,€18.00m
4,Adam Armstrong,eng ENG,"FW,MF",Southampton,27.0,1997.0,20.0,15,1.248,2.0,2.0,4.0,0.14,0.14,€18.00m


In [22]:
def convert_market_value(x):
    if pd.isna(x):
        return None

    x = str(x).strip()
    x = x.replace("€", "").lower()

    if x.endswith("m"):
        return float(x[:-1]) * 1000000
    elif x.endswith("k"):
        return float(x[:-1]) * 1000
    else:
        return float(x)

df["Market Value Numeric"] = df["Market Value"].apply(convert_market_value)

In [23]:
numeric_columns = ["Age", "MP", "Starts", "Min", "Gls", "Ast", "G+A", "Market Value Numeric"]

for col in numeric_columns:
    df[col] = pd.to_numeric(df[col], errors="coerce")

df = df.dropna(subset=numeric_columns)

In [24]:
df[["Gls", "Ast", "G+A", "Market Value Numeric"]].head()

,Gls,Ast,G+A,Market Value Numeric
0,0.0,0.0,0.0,18000000.0
1,1.0,1.0,2.0,6000000.0
2,0.0,0.0,0.0,40000000.0
3,2.0,0.0,2.0,18000000.0
4,2.0,2.0,4.0,18000000.0


In [25]:
X = df[["Age", "MP", "Starts", "Min", "G+A"]]
y = df["Market Value Numeric"]

scaler = StandardScaler()
X_scaled = scaler.fit_transform(X)

X_train, X_test, y_train, y_test = train_test_split(
    X_scaled, y, test_size=0.2, random_state=42
)

model = LinearRegression()
model.fit(X_train, y_train)

predictions = model.predict(X_test)

r2 = r2_score(y_test, predictions)
mae = mean_absolute_error(y_test, predictions)

print("R² Score:", r2)
print("Mean Absolute Error:", mae)

R² Score: 0.21869449015527598
Mean Absolute Error: 15311783.661951607


In [26]:
from sklearn.ensemble import RandomForestRegressor

rf = RandomForestRegressor(random_state=42)
rf.fit(X_train, y_train)

rf_predictions = rf.predict(X_test)

rf_r2 = r2_score(y_test, rf_predictions)

print("Random Forest R² Score:", rf_r2)

Random Forest R² Score: 0.24047644776002974


The linear regression model was used to predict player market values based on performance statistics.

StandardScaler was applied to normalize the variables before training the model.

The Linear Regression model achieved an R² score of approximately 0.22, which means the selected variables explain a limited part of the variation in market value.

The Mean Absolute Error is relatively high, indicating that predictions can differ significantly from actual market values.

To compare model performance, a Random Forest Regressor was also tested.

The Random Forest model achieved a slightly higher R² score than Linear Regression, suggesting that it captured the relationship between performance statistics and market value more effectively.